Toposort Java Refactor (#1679)
cpinitiative/usaco-guide: content/4_Gold/TopoSort.mdx

import java.io.*;

public class CourseSchedule {

> 	public static List<Integer>[] g;

Reviewer: if you're gonna refactors at least rename this to something more explicit like `neighbors`

Reviewer: uh, not resolved?

	public static List<Integer> topo = new ArrayList<Integer>();
	public static int N;
	public static boolean[] visited;
	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		N = Integer.parseInt(st.nextToken());
		int M = Integer.parseInt(st.nextToken());
		g = new List[N];
		visited = new boolean[N];
		for (int i = 0; i < N; i++) {
			g[i] = new ArrayList<Integer>();
		}
		for (int m = 0; m < M; m++) {
			st = new StringTokenizer(br.readLine());
			int a = Integer.parseInt(st.nextToken()) - 1;
			int b = Integer.parseInt(st.nextToken()) - 1;
			g[a].add(b);
		}

		compute();
		int[] ind = new int[N];
		
		for (int i = 0; i < N; i++) ind[topo.get(i)] = i;
		for (int i = 0; i < N; i++)

> 			for (int j: g[i])

Reviewer: ```suggestion
			for (int j : g[i])
```

		Collections.reverse(topo);
	}

> 	public static void dfs(int v) { //not always lexigraphically least

Reviewer 2: ```suggestion
	public static void dfs(int v) { //not always lexicographically least
```

<JavaSection>

```java

> static int[] in_degree;

Reviewer: camelCase? (there's some other places that don't have it, be sure to do that)
also uh wrap this in a class

</LanguageSection>

<Info title="Pro Tip">

> If the array containing the end order does not equal to the number of elements that needed to be sorted, then 

Reviewer 2: ```suggestion
If the length of the array containing the end order does not equal the number of elements that need to be sorted, then 
```
is this what you meant?

> If the array containing the end order does not equal to the number of elements that needed to be sorted, then 
> there is a cycle in the graph constructed by `edges`.

Reviewer 3: ```suggestion
If the array containing the end order does not equal to the number of elements that needed to be sorted, then 
there is a cycle in the graph.
```

//longest_path
public class Main {

> 	static int[] prev_flight, dist, in_degree;

Reviewer: camelCase

		PrintWriter pw = new PrintWriter(System.out);

		ArrayDeque<Integer> answer = new ArrayDeque<>(); //Acts as a Stack

> 		int temp = N-1;

Reviewer 2: ```suggestion
		int temp = N - 1;
```
can you add spaces in all places that need them (can't add suggestions but lines 559, 561, 668, 670 and there's probably more that i didn't find)

		PrintWriter pw = new PrintWriter(System.out);

> 		ArrayDeque<Integer> answer = new ArrayDeque<>(); //Acts as a Stack

Reviewer: ```suggestion
		ArrayDeque<Integer> answer = new ArrayDeque<>();  // Acts as a Stack
```
